pkg/operator/controller/status/controller.go

```diff
@@ -392,24 +413,61 @@ func checkIngressControllersUpgradeableCondition(ingresses []operatorv1.IngressC
 	return false, message
 }
 
+// checkGatewayAPICRDs will check if Gateway API CRDs exist and if they are compatible
+// TODO: Consider getting the managedCRDs from the gatewayAPI controller? Currently duplicates the list from there.
+// TODO: validate existing gatewayapi crds with crd-schema-checker
+func checkGatewayAPICRDs(gatewayAPICRDs []apiextensionsv1.CustomResourceDefinition) (bool, string) {
+	if len(gatewayAPICRDs) == 0 {
+		return true, "No Gateway API CRDs found, upgradeable by default."
+	}
+
+	managedCRDNames := map[string]bool{}
+	for _, crd := range managedGatewayAPICRDs {
+		managedCRDNames[crd.Name] = true
+	}
+
+	nonStandardCRDs := []string{}
+	for _, crd := range gatewayAPICRDs {
+		if !managedCRDNames[crd.Name] {
+			nonStandardCRDs = append(nonStandardCRDs, crd.Name)
+		}
+		// else: crd-schema-checker
+	}
+
+	if len(nonStandardCRDs) > 0 {
+		return false, fmt.Sprintf("Non-standard Gateway API CRDs found: %s", strings.Join(nonStandardCRDs, ", "))
+	}
+
+	return true, ""
+}
+
 // computeOperatorUpgradeableCondition computes the operator's Upgradeable
 // status condition.
-func computeOperatorUpgradeableCondition(ingresses []operatorv1.IngressController) configv1.ClusterOperatorStatusCondition {
-	ingressControllersUpgradeable, message := checkIngressControllersUpgradeableCondition(ingresses)
+func computeOperatorUpgradeableCondition(ingresses []operatorv1.IngressController, gatewayAPICRDs []apiextensionsv1.CustomResourceDefinition) configv1.ClusterOperatorStatusCondition {
+	ingressControllersUpgradeable, ingressControllersUpgradeableMessage := checkIngressControllersUpgradeableCondition(ingresses)
+	gatewayAPICRDsCompatible, gatewayAPICRDsMessage := checkGatewayAPICRDs(gatewayAPICRDs)
 
-	if ingressControllersUpgradeable {
+	if ingressControllersUpgradeable && gatewayAPICRDsCompatible {
 		return configv1.ClusterOperatorStatusCondition{
 			Type:   configv1.OperatorUpgradeable,
 			Status: configv1.ConditionTrue,
-			Reason: "IngressControllersUpgradeable",
+			Reason: "AllComponentsUpgradeable", // TODO: Find better description
 		}
 	}
 
+	messages := []string{}
+	if !ingressControllersUpgradeable {
+		messages = append(messages, ingressControllersUpgradeableMessage)
+	}
+	if !gatewayAPICRDsCompatible {
+		messages = append(messages, gatewayAPICRDsMessage)
+	}
+
 	return configv1.ClusterOperatorStatusCondition{
 		Type:    configv1.OperatorUpgradeable,
 		Status:  configv1.ConditionFalse,
-		Reason:  "IngressControllersNotUpgradeable",
-		Message: message,
+		Reason:  "ComponentsNotUpgradeable",
```

Comment:
OTOH it would be better to have distinct reasons here. `IngressControllersNotUpgradeable`, `GatewayAPICRDsNotCompatible`, `MultipleProblems`/`MultipleComponentsNotUpgradeable`.